Sources/Sentry/Public/SentryOptions.h

```diff
@@ -191,6 +191,19 @@ NS_SWIFT_NAME(Options)
  */
 @property (nonatomic, assign) BOOL enableAutoPerformanceTracing;
 
+/**
+ * A block that creates the initial scope to use when starting the SDK.
+ * @note Return @c nil to allow the SDK to lazily create its own scope.
+ * @note The default simply returns @c nil .
+ */
+@property (nonatomic) SentryScope * _Nullable (^initialScopeFactory)(void);
+
+/**
+ * Provides an initial scope to use when starting the SDK.
+ * @param callback A block that configures a fresh scope as desired.
+ */
+- (void)initialScope:(void (^)(SentryScope *scope))callback;
```

Comment:
I wonder whether we should rename this to `-configureInitialScope:` and rename `initialScopeFactory` to `initialScope` — @philipphofmann what do you think?